src/components/AnalyticsProvider/index.tsx

```diff
@@ -1,51 +1,42 @@
-import * as React from 'react';
+import React, {useEffect, useMemo, ReactNode} from 'react';
 
 import AnalyticsProviderContext from '../../contexts/AnalyticsProviderContext';
-import {UnknownRecord} from '../../types/common';
-
-interface Props {
-  onInitialize(): void;
-  onPageView?(params?: UnknownRecord): void;
-  onEvent?(name: string, params?: UnknownRecord): void;
-  onClick?(name: string, params?: UnknownRecord): void;
-  onSet?(...args: [string, UnknownRecord] | [UnknownRecord]): void;
-  onSetUserId?(userId: string | null): void;
-  onSetUserProperty?(params: UnknownRecord): void;
-  onException(params?: UnknownRecord): void;
-  children: React.ReactNode;
+import {Analytics} from '../../mixin/analytics';
+import {AnalyticsClient} from '../../interfaces';
+import {SetupParams} from '../../types';
+
+interface CustomSetupProps {
+  children: ReactNode;
+  client: AnalyticsClient;
+}
+
+interface DefaultSetupProps {
```

Comment:
```suggestion
export interface CustomSetupProps {
  children: ReactNode;
  client: AnalyticsClient;
}

export interface DefaultSetupProps {
```

사족) 각 prop interface도 export해줘도 좋을 것 같습니다 ㅎㅎ..